apps/ui/src/apis/communities/fetchAllCommunities.ts

```diff
@@ -1,60 +1,38 @@
 import { Community } from '@communecar/types';
 import { graphqlRequest } from '../graphql';
-
-interface UserNode {
-  profileImage: string | null;
-}
-
-interface UserCommunityNode {
-  userByUserId: UserNode;
-}
-
-interface CommunityNode {
-  id: number;
-  title: string;
-  description: string;
-  userCommunitiesByCommunityId: {
-    nodes: UserCommunityNode[];
-  };
-}
-
-interface CommunitiesData {
-  allCommunities: {
-    nodes: CommunityNode[];
-  };
-}
+import { getFetchAllCommunitiesQuery } from '../utils/communitiesQueries';
+import { AllCommunitiesData } from '../types/communitiesResponse';
+import { locationExtraction } from '../location/location';
 
 const fetchAllCommunities = async (): Promise<Community[]> => {
-  const query = `
-    query {
-      allCommunities {
-        nodes {
-          id
-          title
-          description
-          userCommunitiesByCommunityId {
-            nodes {
-              userByUserId {
-                profileImage
-              }
-            }
-          }
-        }
-      }
-    }`;
-
-  const data = await graphqlRequest<CommunitiesData>(query);
-
-  return data.allCommunities.nodes.map((node): Community => {
-    const picturesUrl = node.userCommunitiesByCommunityId.nodes
-      .map((userCommunity) => userCommunity.userByUserId.profileImage)
-      .filter((url): url is string => url != null);
-    return {
-      ...node,
-      numberOfMembers: node.userCommunitiesByCommunityId.nodes.length,
-      picturesUrl,
-    };
-  });
+  const query = getFetchAllCommunitiesQuery();
+
+  const data = await graphqlRequest<AllCommunitiesData>(query);
+
+  const allCommunities = data.allCommunities.nodes.map(
+    async (node): Promise<Community> => {
+      const location =
+        node.lat && node.long ? await locationExtraction(node) : undefined;
+
+      const picturesUrl = node.userCommunitiesByCommunityId.nodes
+        .map((userCommunity) => userCommunity.userByUserId.profileImage)
+        .filter((url): url is string => url != null);
```

Comment:
(url: string) => url !== null) or !isNil(url)

Reply:
It doesn't work any other way